clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainStorageEngine.java

```diff
@@ -175,6 +339,56 @@ public boolean shrink() {
     return storage.shrink();
   }
 
+  protected ByteBuffer getExtensionHeader(long xtnAddress) {
+    checkExtensionHeaderExists();
+    return storage.readBuffer(xtnAddress, extendedChainHeaderSize);
+  }
+
+  protected WriteContext getExtensionWriteContext(long xtnAddress) {
+    checkExtensionHeaderExists();
+    return new WriteContext() {
+
+      @Override
+      public void setLong(int offset, long value) {
+        if (offset < 0 || offset >= extendedChainHeaderSize) {
+          throw new IllegalArgumentException("Offset not within bounds 0 >= " + offset + " < " + extendedChainHeaderSize);
+        } else {
+          storage.writeLong(xtnAddress + offset, value);
+        }
+      }
+
+      @Override
+      public void flush() {
+        //no-op
+      }
+    };
+  }
+
+  protected long chainAddressToExtensionAddress(long chainAddress) {
```

Comment:
Why does this method and it's counterpart need to be exposed?  Shouldn't where the extension header gets stored be a concern only for *this* class and not for the subclass that just needs to write and read the header, not care about where it gets put.